### DISCUSSION TITLE
Potential Bug with dls

### DISCUSSION THREAD

**Discussion by EnsonoStevenPerva** [IBM/zoau-samples#31] [General] [answered]
Some component of ZOAU's `dls` utility seems to ignore hyphens in the DSN. It is causing some problems, especially when trying to use other switches with the command. It appears the hyphen is completely ignored in the DSN. Here are the steps to reproduce:

1. Allocate a dataset with a hyphen in the name (i.e. `dtouch IBMUSER.JCL-LIB`)
2. Perform a list of the HLQ you just created (i.e. `dls IBMUSER.*`)
	2a. Observe your list includes the dataset but without the hyphen:
			`IBMUSER.JCLLIB`
3. Perform a list with additional options (i.e. `dls -us IBMUSER.*`)
	3a. Observe the error presented when the utility attempts to pull additional information:
		`BGYSC2006I Unable to obtain dataset information for dataset IBMUSER.JCLLIB on volume TSOVOL.`

**Comment by MikeFultonDev** [comment, marked as answer]
Verified with latest ZOAU. This is a bug. Thank you for reporting it @EnsonoStevenPerva . @ddimatos fyi

**Comment by ddimatos** [comment]
Thank you for reporting this, it  has been captured in JIRA NAZARE-8850 and will be reviewed.
